src/analyze/analyze-condition.c

```diff
@@ -123,12 +123,15 @@ static int verify_conditions(char **lines, RuntimeScope scope, const char *unit,
                 }
         }
 
-        r = condition_test_list(u->asserts, environ, assert_type_to_string, log_helper, u);
-        if (u->asserts)
+        condition_test_logger_t logger = arg_quiet ? NULL : log_helper;
+        condition_to_string_t to_string = arg_quiet ? NULL : assert_type_to_string;
```

Comment:
hmm, I think the `assert(!!logger == !!to_string);` in condition_test_list() should be dropped and then the `assert_type_to_string` function be passed always. that's just a lot simpler, and it doesn't matter if condition_test_list() actually uses the to_string function or not...